### PR TITLE
maxas-limit property

### PR DESCRIPTION
**Pull request by andy91315**
1. do not know maxas-limit default value
2. ruby test 5 failure and 1 error not related to this property
3. static analysis error: Class has too many lines

### CODE REVIEW

File: lib/cisco_node_utils/command_reference_common_bgp.yaml [hunk 1]
```diff
@@ -100,6 +100,11 @@ bgp:
     config_set_append: '<state> log-neighbor-changes'
     default_value: false
 
+  maxas_limit:
+    config_get_token_append: '/^maxas-limit (\d+)$/'
+    config_set_append: '<state> maxas-limit <limit>'
+    default_value: 50
```
Comment (chrisvanheuveln) [inline lib/cisco_node_utils/command_reference_common_bgp.yaml:106]: Where does 50 come from? Your comments indicated that there was no default value. When I configure 'maxas-limit 50' it nvgens (so it's not the default value). Use a tilde to indicate nil instead:

```
  default_value: ~
```

File: tests/test_router_bgp.rb [hunk 2]
```diff
@@ -566,6 +566,36 @@ def test_routerbgp_default_neighbor_fib_down_accelerate
...
+                   "should be set to default value of '50'")
+      bgp.destroy
+    end
+  end
+
+  def test_routerbgp_default_maxas_limit
+    asnum = 55
+    bgp = RouterBgp.new(asnum)
+    assert_equal(50, bgp.maxas_limit,
+                 "bgp maxas-limit default value should be '50'")
```
Comment (chrisvanheuveln) [inline tests/test_router_bgp.rb:595]: For this test you should:
1. set maxas to some non default value and check the result
2. set maxas to default using the default api, then check the result
   
   e.g:

```
  bgp.maxas_limit = 3
  assert_equal(3, bgp.maxas_limit, ...
  bgp.maxas_limit = bgp.default_maxas_limit
  assert_equal(bgp.default_maxas_limit, bgp.maxas_limit, ...
```

**Comment by chrisvanheuveln** [comment]
Please elaborate on the test failures you're seeing. You shouldn't be getting any minitest failures. For the rubocop failure make this change to test_router_bgp.rb:

```
 # TestRouterBgp - Minitest for RouterBgp class
+# rubocop:disable Metrics/ClassLength
 class TestRouterBgp < CiscoTestCase
```

**Comment by glennmatthews** [comment]
Rather than disabling the Metrics/ClassLength limit altogether for TestRouterBgp, as Chris suggest, I would recommend updating `.rubocop.yml` with whatever the new limit needs to be - something like:

``` diff
 Metrics/ClassLength:
-  Max: 722
+  Max: 743
```

**Comment by chrisvanheuveln** [comment]
@glennmatthews I'm not sure I see the value in that. You're disabling the cop either way. There are more   properties coming for this provider so this will have to be changed again for the next ones.

**Comment by glennmatthews** [comment]
If we don't see any value in the Metrics/ClassLength cop, we should just turn it off globally - rather than disabling it for each individual file that happens to pass some arbitrary limit.

**Comment by chrisvanheuveln** [comment]
Glenn and I discussed this offline and agreed to disable it globally. Andi, please make this change along with your diff:

```
diff --git a/.rubocop.yml b/.rubocop.yml
index a94e12d..8ac8011 100644
--- a/.rubocop.yml
+++ b/.rubocop.yml
@@ -4,9 +4,6 @@
 Metrics/AbcSize:
   Max: 62

-Metrics/ClassLength:
-  Max: 722
-
 Metrics/CyclomaticComplexity:
   Max: 17

@@ -28,6 +25,10 @@ Metrics/PerceivedComplexity:
 # Team preferences different from Rubocop defaults:
 #

+# Test files and large providers can get very large
+Metrics/ClassLength:
+  Enabled: false
+
```

**Pull request closed without merging**